app/src/userpages/components/ShareSidebar/UserList.jsx

```diff
@@ -0,0 +1,67 @@
+import React, { useCallback } from 'react'
+import styled from 'styled-components'
+import Sidebar from '$shared/components/Sidebar'
+import UserPermissions from './UserPermissions'
+import ErrorMessage from './ErrorMessage'
+
+const UnstyledUserList = ({
+    currentUser,
+    items,
+    onSelect,
+    permissions,
+    removeUser,
+    resourceType,
+    selectedUserId,
+    updatePermission,
+    userErrors,
+    ...props
+}) => {
+    const onClick = useCallback((e) => {
+        if (e.target === e.currentTarget) {
+            // Select none on click background.
+            onSelect()
+        }
+    }, [onSelect])
+
+    return (
+        // eslint-disable-next-line jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events
+        <div
+            {...props}
+            onClick={onClick}
+        >
+            {items.map(([userId, userPermissions]) => (
+                <React.Fragment key={userId}>
+                    <Sidebar.Container
+                        as={UserPermissions}
+                        invalid={!!userErrors[userId]}
+                        isCurrentUser={currentUser === userId}
+                        isSelected={selectedUserId === userId}
+                        onSelect={onSelect}
+                        permissions={permissions}
+                        removeUser={removeUser}
+                        resourceType={resourceType}
+                        updatePermission={updatePermission}
+                        userId={userId}
+                        userPermissions={userPermissions}
+                    />
+                    {!!userErrors[userId] && (
+                        <Sidebar.Container as={ErrorMessage}>
+                            {userErrors[userId].message}
+                        </Sidebar.Container>
+                    )}
```

Comment:
Here's something new. `UserPermissions` became `Sidebar.Container` thus they come with padding. Rendering errors within that structure wouldn't end up well (noticable padding around a red error box). I render errors outside (after) `UserPermissions` then. Btw, `ErrorMessage` is also a `Sidebar.Container` here!